lib/actions/parse_csv.ex

```diff
@@ -0,0 +1,70 @@
+defmodule WorkflowEngine.Actions.ParseCsv do
+  @moduledoc """
+  Executes a request against the internal API using BXDK.
+  """
+  use OK.Pipe
+
+  @behaviour WorkflowEngine.Action
+
+  @impl WorkflowEngine.Action
+
+  NimbleCSV.define(CSVComma,
+    separator: ",",
+    escape: "\""
+  )
+
+  NimbleCSV.define(CSVSemiColon,
+    separator: ";",
+    escape: "\""
+  )
+
+  NimbleCSV.define(CSVTab,
+    separator: "\t",
+    escape: "\""
+  )
+
+  def execute(state, %{"type" => "parse_csv", "csv_settings" => csv_settings} = step) do
+    {:ok, data} = get_data(step, state)
```

Comment:
This will crash if the `get_data` returns an error tuple, which it does if data is not a map/binary or if it doesn't exist.
While the "let it crash" strategy is sometimes used in Elixir (especially in Tasks), in this case as we already kind of handle the error with the error tuple downstairs, I would personally catch it and log it.